test/controllers/foreman_register/hosts_controller_test.rb

```diff
@@ -75,6 +75,28 @@ class HostsControllerTest < ActionController::TestCase
           assert_response :not_found
         end
       end
+
+      context 'with error' do
+        it 'no OS' do
+          get :register, params: { token: FactoryBot.create(:host).registration_token }
+          assert_response :bad_request
+          assert_includes @response.body, "[Foreman::Exception]: Must provide an operating system\"\nexit 1\n"
```

Comment:
The message is a bit confusing, it suggests user should provide that e.g. as a parameter. It should be saying the host does not have OS associated. I know it's not added in this PR but could we change it in here, before we hardcode it in our test suite?